src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/CollectionsMarshalTests.cs

```diff
@@ -0,0 +1,151 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class CollectionsMarshalTests
+    {
+        [Fact]
+        public unsafe void NullListAsSpanValueType()
+        {
+            List<int> list = null;
+            Span<int> span = CollectionsMarshal.AsSpan(list);
+
+            Assert.Equal(0, span.Length);
+
+            fixed (int* pSpan = span)
+            {
+                Assert.True(pSpan == null);
+            }
+        }
+
+        [Fact]
+        public unsafe void NullListAsSpanClass()
+        {
+            List<object> list = null;
+            Span<object> span = CollectionsMarshal.AsSpan(list);
+
+            Assert.Equal(0, span.Length);
+        }
+
+        [Fact]
+        public void ListAsSpanValueType()
+        {
+            var list = new List<int>();
+            foreach (int length in Enumerable.Range(0, 36))
+            {
+                list.Clear();
+                ValidateContentEquality(list, CollectionsMarshal.AsSpan(list));
+
+                for (var i = 0; i < length; i++)
```

Comment:
nit: use of var in for loop indexer.
```suggestion
                for (int i = 0; i < length; i++)
```